internal/set/set.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package set
+
+type Set[T comparable] map[T]struct{}
+
+func SliceToSet[T comparable](sl []T) Set[T] {
+	s := Set[T]{}
+	for _, v := range sl {
+		s.Add(v)
+	}
+	return s
+}
+
+func MapToSet[K comparable, V any](m map[K]V) Set[K] {
```

Comment:
This name kind of bugs me, because I wasn't sure when looking at the callsite whether it was a set of the keys in the map or a set of the values. How about set.KeySet ? I know that this loses parallelism with SliceToSet, but I feel like that's a small price to pay.

Weak suggestion, feel free to ignore.

Reply:
I probably wouldn't change the name of the type to `set.KeySet` since the `Set` could be made of anything not just map keys. I could change this function name to `MapKeysToSet`?

Reply:
huh? if you called this function on a map, the resulting set can't be made of anything, it is specifically made of the keys of the given map. 

oh you think I wass suggesting changing the name of the type? no, I Iwas just suggesting changing the name of ths function.

MapKeysToSet is all rght, i still prefer KeySet, but it's up to you.

Reply:
Yeah that works, and maps with the Effective Go practice of not using the package name in exported members from the package. Will do that.

Reply:
Done in [8c8978f](https://github.com/GoogleCloudPlatform/ops-agent/pull/901/commits/8c8978ffc91e2e1e2dcc118c3017717a6d266f39)